### PR TITLE
Mmt 3551: Setup permittedUser for different Env

### PR DESCRIPTION
**Pull request by tle1989**

### CODE REVIEW

File: static/src/js/utils/getPermittedUser.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ *Logic to determine permittedUser based on environment
+ */
+const getPermittedUser = () => {
+  // For development environment
+  if (process.env.NODE_ENV === 'development') {
```
Comment (cgokey) [inline static/src/js/utils/getPermittedUser.js:6]: to be consistent with other apps, you'll want to use check AppConfig version instead.   We talked about writing a ticket for future to possibly move things over to using process.env.NODE_ENV

File: static/src/js/utils/getPermittedUser.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    return 'typical'
+  }
+
+  // For SIT environment
+  return process.env.REACT_APP_SIT_PERMITTED_USER
```
Comment (cgokey) [inline static/src/js/utils/getPermittedUser.js:11]: You should return `user.uid` here, so you'll want to pass in user to the function to destructure out the uid.

File: static/src/js/__tests__/App.test.js [hunk 3]
```diff
@@ -87,9 +87,14 @@ jest.mock('../components/Layout/Layout', () => {
   })
 })
 
+jest.mock('../utils/getPermittedUser', () => ({
```
Comment (cgokey) [inline static/src/js/__tests__/App.test.js:90]: I don't think you'll need this if you use App Config version

File: static/src/js/App.jsx [hunk 4]
```diff
@@ -40,6 +41,17 @@ const Redirects = redirectKeys.map(
   )
 )
 
+// const getPermittedUser = () => {
```
Comment (cgokey) [inline static/src/js/App.jsx]: remove comments

Comment (tle1989) [reply static/src/js/App.jsx]: Removed

File: static/src/js/utils/getPermittedUser.js (hunk 2 above)
Comment (tle1989) [reply static/src/js/utils/getPermittedUser.js]: updated this

File: static/src/js/__tests__/App.test.js (hunk 3 above)
Comment (tle1989) [reply static/src/js/__tests__/App.test.js]: Removed, thanks!

File: static/src/js/utils/getPermittedUser.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ */
+const getPermittedUser = (user) => {
+  const { version } = getApplicationConfig()
+  // For development environment
+  if (version === 'development') {
+    return 'typical'
+  }
+
+  // For SIT environment
+  return user?.uid
```
Comment (cgokey) [inline static/src/js/utils/getPermittedUser.js:15]: looks good, just destructure uid

File: static/src/js/utils/__tests__/getPermittedUser.test.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should return "typical" for development environment', () => {
+    // Mock the application config to return 'development' version
+    getApplicationConfig.mockReturnValue({ version: 'development' })
+
+    // Set environment to development
+    process.env.NODE_ENV = 'development'
```
Comment (cgokey) [inline static/src/js/utils/__tests__/getPermittedUser.test.js:18]: not needed

File: static/src/js/utils/__tests__/getPermittedUser.test.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+    expect(getPermittedUser()).toEqual('typical')
+  })
+
+  it('should return the user\'s uid for non-development environment', () => {
+    // Mock the application config to return 'production' version
+    getApplicationConfig.mockReturnValue({ version: 'production' })
+
+    // Set environment to production
+    process.env.NODE_ENV = 'production'
```
Comment (cgokey) [inline static/src/js/utils/__tests__/getPermittedUser.test.js:28]: no needed

File: static/src/js/__tests__/App.test.js [hunk 8]
```diff
@@ -236,6 +236,7 @@ describe('App component', () => {
       await waitFor(() => {
         expect(screen.getByTestId('mock-manage-cmr-page')).toBeInTheDocument()
         expect(ManageCmrPage).toHaveBeenCalledTimes(3)
+        expect(ManageCmrPage).toHaveBeenCalledTimes(3)
```
Comment (cgokey) [inline static/src/js/__tests__/App.test.js:239]: dup line

File: static/src/js/utils/getPermittedUser.js (hunk 5 above)
Comment (tle1989) [reply static/src/js/utils/getPermittedUser.js:15]: Done

File: static/src/js/utils/__tests__/getPermittedUser.test.js (hunk 7 above)
Comment (tle1989) [reply static/src/js/utils/__tests__/getPermittedUser.test.js:28]: Done

File: static/src/js/__tests__/App.test.js (hunk 8 above)
Comment (tle1989) [reply static/src/js/__tests__/App.test.js:239]: Removed

**Review by dmistry1 — APPROVED** [review]

**Review by cgokey — APPROVED** [review]

**Pull request merged**